services/beehive-upload-server/user-service/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+// testing: docker run -d --name upload-server -p 8080:80  waggle/beehive-upload-server
+// curl localhost:8080/user
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/gorilla/mux"
+)
+
+type NodeObj struct {
+	ID string `json:"id"`
+}
+
+type APIResponse struct {
+	Data interface{} `json:"data"`
+}
+
+type APIErrorResponse struct {
+	Error string `json:"error"`
+}
+
+func GetUsers() (users []string, err error) {
+
+	file, err := os.Open("/etc/passwd")
+
+	if err != nil {
+		err = fmt.Errorf("error opening file: %s", err.Error())
+		return
+	}
+
+	defer file.Close()
+
+	reader := bufio.NewReader(file)
```

Comment:
You may like using bufio.NewScanner for this. It makes it easy to write a "for each line via":
```go
scanner := bufio.NewScanner(file)

for scanner.Scan() {
    // process line in scanner.Text()
}

if scanner.Err() != nil {
  handle / return scanner.Err()
}
```

Reply:
thanks for the tip !